### ISSUE TITLE
javascript: es2015 and es2017 functions, docblockr does not create template

### ISSUE THREAD

**Issue by chrishiestand** [nikhilkalige/docblockr#203]
Take this function for example:

```javascript

function setMetadata(file, headers, extra) {
}
```

I can get a helpful autocomplete if I start typing a doc string above it:

```javascript
/**
 * [setMetadata description]
 * @method setMetadata
 * @param  {[type]}    file    [description]
 * @param  {[type]}    headers [description]
 * @param  {[type]}    extra   [description]
 */
function setMetadata(file, headers, extra) {
}
```

But this fails to work with `export function` (adopted in ecmascript 2015), for example:
```javascript
/**
 *  
 */
export function setMetadata(file, headers, extra) {
}
```

This is the same with the following combinations:

```javascript
export default function setMetadata(file, headers, extra) {
}

export function setMetadata(file, headers, extra) {
}

export async function setMetadata(file, headers, extra) {
}

export default async function setMetadata(file, headers, extra) {
}
```

Note that if the function is declared an `async` function it always returns a `Promise` object (which would be very helpful to add `@returns` to auto-complete). `async/await` was recently [moved to `stage 4` so it is likely to be accepted into the next ecmascript standard](https://github.com/tc39/proposals/blob/master/finished-proposals.md) (and it is currently used with transpilers like babel).

**Comment by jwhitmarsh** [comment]
This would be awesome - does anyone know if this is "newbie" appropriate, because I'd be happy to take a look at it?

**Comment by chrishiestand** [comment]
Nice! Yeah I was going to say that it should be pretty straight forward. Adding a `@returns` in the case of `async` may be more challenging?

Take a look at one of the other language parsers which require return types like c++
```
CppParser.prototype.get_function_return_type = function(name, retval) {
    return ((retval != 'void') ? retval : null);
};
```

Maybe that function prototype can be added to the js parser with async detection inside?

**Comment by jwhitmarsh** [comment]
Just did a quick look through this and `get_function_return_type` receives the function name (i.e `setMetaData`) without any of the function decoration. I haven't got time right now, but over the weekend I'll check to see if this can be updated. 

Is anyone using dockblockr with C++ who can chip in about whether `static|private` (etc) decorated methods have different templates? Because if so then that gives us something to work with...

**Comment by MoritzKn** [comment]
C++ is adding the `return` type by returning it as the third element of the array [L43](https://github.com/nikhilkalige/docblockr/blob/master/lib/languages/cpp.js#L43) and then evaluating it in the [get_function_return_type](https://github.com/nikhilkalige/docblockr/blob/master/lib/languages/cpp.js#L75) function. You can do the same, but should ([like PHP](https://github.com/nikhilkalige/docblockr/blob/master/lib/languages/php.js#L140)) call the super function as fall back.
Ask if you have further questions.

**Comment by MoritzKn** [comment]
@jwhitmarsh's PR is merged and released with version 0.8.4.
This implemented the described functionality.
